src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -180,10 +180,10 @@ static int get_sdr_type(enum xcl_group_kind kind)
 	return type;
 }
 
-static void update_cache_expiry_time(struct xocl_hwmon_sdm *sdm)
+static void update_cache_expiry_time(struct xocl_hwmon_sdm *sdm, uint8_t repo_id)
 {
-	sdm->cache_expires = ktime_add(ktime_get_boottime(),
-                                   ktime_set(sdm->cache_expire_secs, 0));
+	sdm->cache_expires[repo_id] = ktime_add(ktime_get_boottime(),
+                                      ktime_set(sdm->cache_expire_secs, 0));
```

Comment:
What guarantee is there that repo_id is with the range of the array?

Reply:
Hi Soren, 

While parsing sensor buffer in hwmon_sdm driver, it defined  sdr_get_id() API which takes repo_type input and returns "repo_id" value which is in the range of [0, SDR_TYPE_MAX). This same value is being passed to this update_cache_expiry_time() also. 

For safe side, I added a if condition for repo_id in hwmon_sensor_show() if it is in valid range or not. Please review it again, thank you.